conformance/tests/baseline-admin-network-policy-experimental-egress-rules.go

```diff
@@ -111,23 +111,23 @@ var CNPBaselineTierEgressNodePeers = suite.ConformanceTest{
 		require.NoErrorf(t, err, "unable to fetch the server pod")
 		t.Run("Should support an 'allow-egress' rule policy for egress-node-peer", func(t *testing.T) {
 			// harry-potter-0 is our client pod in gryffindor namespace
-			// ensure egress is ALLOWED to forbidden-forrest from gryffindor at the 36363 TCP port
+			// ensure egress is ALLOWED to forbidden-forrest from gryffindor at the s.HostNetworkPortRangeStart TCP port
 			// egressRule at index0 should take effect
 			success := kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-0", "tcp",
-				serverPod.Status.PodIP, int32(36363), s.TimeoutConfig.RequestTimeout, true)
+				serverPod.Status.PodIP, int32(s.HostNetworkPortRangeStart), s.TimeoutConfig.RequestTimeout, true)
 			assert.True(t, success)
 			success = kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-1", "tcp",
-				serverPod.Status.PodIP, int32(36364), s.TimeoutConfig.RequestTimeout, true) // Pass rule at index2 takes effect
+				serverPod.Status.PodIP, int32(s.HostNetworkPortRangeStart+1), s.TimeoutConfig.RequestTimeout, true) // Pass rule at index2 takes effect
 			assert.True(t, success)
 		})
 		t.Run("Should support a 'deny-egress' rule policy for egress-node-peer", func(t *testing.T) {
 			// harry-potter-1 is our client pod in gryffindor namespace
 			// ensure egress is DENIED to rest of the nodes from gryffindor; egressRule at index1 should take effect
 			success := kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-1", "udp",
-				serverPod.Status.PodIP, int32(34346), s.TimeoutConfig.RequestTimeout, false)
+				serverPod.Status.PodIP, int32(s.HostNetworkPortRangeStart+4), s.TimeoutConfig.RequestTimeout, false)
 			assert.True(t, success)
 			success = kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-1", "sctp",
-				serverPod.Status.PodIP, int32(9003), s.TimeoutConfig.RequestTimeout, false)
+				serverPod.Status.PodIP, int32(s.HostNetworkPortRangeStart+6), s.TimeoutConfig.RequestTimeout, false)
```

Comment:
all this "+6" etc is very non-obvious, but I guess the original hard-coded port numbers were non-obvious too...

(could possibly be `s.HostNetworkPorts[6]` to make it match the yaml better though?)

Reply:
oh wow you can see I was interrupted a 100 times while writing this (sorry for that, this PR is a bit of a mess)
I think I started with the list indexing, but then decided that no one will pass the full list of ports to use and it may be a better API to just pass `HostNetworkPortRangeStart` (also potentially no change will be required when we need to use more ports as long as the end of the range doesn't intersect with anything)
So what I wanted to do is to use `s.HostNetworkPortRangeStart+6` both in code and yamls, but lmk if you think list is better